packages/core/src/client/ClientService.ts

```diff
@@ -67,7 +63,7 @@ export class ClientService {
   public async deleteClient(clientId: string, password?: string): Promise<unknown> {
     const userId: QualifiedId = {id: this.apiClient.userId as string, domain: this.apiClient.domain || ''};
     await this.backend.deleteClient(clientId, password);
-    return this.database.deleteClient(this.cryptographyService.constructSessionId(userId, clientId));
+    return this.database.deleteClient(constructSessionId({userId, clientId}));
```

Comment:
This change is not equivalent since the `useQualifiedId` is going to be out of sync. 
You will need to inject the `useQualifiedId` in the class and in the `constructSessionId` as well